buildpacks/ruby/tests/integration_test.rs

```diff
@@ -57,9 +62,26 @@ fn test_default_app_ubuntu20() {
 
 #[test]
 #[ignore = "integration test"]
-fn test_default_app_latest_distro() {
+fn test_default_app_ubuntu22() {
     TestRunner::default().build(
         BuildConfig::new("heroku/builder:22", "tests/fixtures/default_ruby"),
+        |context| {
+            println!("{}", context.pack_stdout);
+            assert_contains!(context.pack_stdout, "# Heroku Ruby Buildpack");
+            assert_contains!(
+                context.pack_stdout,
+                r#"`BUNDLE_BIN="/layers/heroku_ruby/gems/bin" BUNDLE_CLEAN="1" BUNDLE_DEPLOYMENT="1" BUNDLE_GEMFILE="/workspace/Gemfile" BUNDLE_PATH="/layers/heroku_ruby/gems" BUNDLE_WITHOUT="development:test" bundle install`"#);
+
+            assert_contains!(context.pack_stdout, "Installing webrick");
+        },
+    );
+}
+
+#[test]
+#[ignore = "integration test"]
+fn test_default_app_latest_distro() {
+    TestRunner::default().build(
+        BuildConfig::new("heroku/builder:24", "tests/fixtures/default_ruby"),
```

Comment:
This test fails for me locally, unless I add `.target_triple("aarch64-unknown-linux-musl")` because it's always compiling the buildpack for amd64 even if run on an arm64 host (and for `heroku/builder:24` if the host is arm64, then the arm64 image will be pulled by default).

Does this pass for you locally? If so, it might mean you have a global `DOCKER_DEFAULT_PLATFORM` set, which is masking the problem (I wouldn't set this globally in your shell).

To resolve the issue, you'll need something like:
https://github.com/heroku/buildpacks-python/blob/de378063c83e1eec6302e79f93f189162b5bbbbf/tests/mod.rs#L31-L38

(Having CI testing of arm64 would have caught this)

Reply:
Updated to use similar logic https://github.com/heroku/buildpacks-ruby/pull/284. I verify that this test now passes for me locally.

I also opened https://github.com/heroku/libcnb.rs/issues/825.